backend/Watchdog.Core/Watchdog.Core.BLL/Services/MemberService.cs

```diff
@@ -1,30 +1,94 @@
-using Watchdog.Core.Common.DTO.Member;
+using Watchdog.Core.Common.DTO.Members;
 using Watchdog.Core.DAL.Context;
 using Watchdog.Core.DAL.Entities;
 
 namespace Watchdog.Core.BLL.Services
 {
+
     public class MemberService : BaseService, IMemberService
     {
-        public MemberService(WatchdogCoreContext context, IMapper mapper) : base(context, mapper)
+        private readonly IEmailSendService _emailSendService;
+
+        public MemberService(WatchdogCoreContext context, IMapper mapper, IEmailSendService emailSendService) : base(context, mapper)
         {
+            _emailSendService = emailSendService;
         }
 
-        public async Task<ICollection<MemberDto>> GetMembersByOrganizationIdAsync(int id)
+        public async Task<Response> InviteMemberAsync(int id)
         {
-            return _mapper.Map<ICollection<MemberDto>>(await _context.Members.Where(m => m.OrganizationId == id).Include(m => m.User).ToListAsync());
+            var member = await _context.Members
+                .Include(m => m.User)
+                .FirstOrDefaultAsync(m => m.Id == id) ?? throw new KeyNotFoundException("Member doesn't exist");
+
+            return await InviteMemberAsync(_mapper.Map<MemberDto>(member));
         }
 
-        public async Task<ICollection<MemberDto>> GetAllMembersAsync()
+        public Task<Response> InviteMemberAsync(MemberDto memberDto)
+        {
+            ExampleTemplateData data = new()
+            {
+                Name = memberDto.User.FirstName,
+                Subject = "Invitation letter"
+            };
+            return _emailSendService.SendAsync(memberDto.User.Email, data);
+        }
+
+        public async Task<MemberDto> AddMemberAsync(NewMemberDto memberDto)
         {
-            var members = await _context.Members.ToListAsync();
+            var user = (await _context.Users.FirstOrDefaultAsync(u => u.Id == memberDto.UserId)) ?? throw new KeyNotFoundException("User doesn't exist");
+            var organization = (await _context.Organizations.FirstOrDefaultAsync(o => o.Id == memberDto.OrganizationId) ?? throw new KeyNotFoundException("Such organization doesn't exist"));
+
+            await _context.Entry(organization).Collection(o => o.Members).LoadAsync();
+
+
+            if (organization.Members.Any(m => m.UserId == user.Id))
+            {
+                throw new ArgumentException("This user already in organization");
+            }
+
+            var member = _mapper.Map<Member>(memberDto);
+
+            foreach (int teamId in memberDto.TeamIds)
+            {
+                member.TeamMembers.Add(new TeamMember { TeamId = teamId});
+            }
+
+            await _context.Members.AddAsync(member);
+
+            await _context.SaveChangesAsync();
+
+            await _context.Entry(member).Reference(m => m.Role).LoadAsync();
+            await _context.Entry(member).Collection(m => m.TeamMembers).Query().Include(tm => tm.Team).LoadAsync();
```

Comment:
Why do you need these lines?

Reply:
It is like .Include(m => ...), but we use that methods when we need to include data to single object (not DbSet) and then teams and roles used by mappper to map member and dto.

Reply:
It looks strange. So please rewrite it